include/fastdds/rtps/transport/TransportInterface.h

```diff
@@ -50,37 +53,49 @@ class TransportInterface
 public:
 
     /**
-    * Aside from the API defined here, an user-defined Transport must define a descriptor data type and a constructor that
-    * expects a constant reference to such descriptor. e.g:
-    *
-    * class MyTransport:
-    * public:
-    *    typedef struct { ... } MyTransportDescriptor;
-    *    MyTransport(const MyTransportDescriptor&);
-    *    ...
-    */
+     * Aside from the API defined here, an user-defined Transport must define a descriptor data type and a constructor that
+     * expects a constant reference to such descriptor. e.g:
+     *
+     * class MyTransport:
+     * public:
+     *    typedef struct { ... } MyTransportDescriptor;
+     *    MyTransport(const MyTransportDescriptor&);
+     *    ...
+     */
     virtual ~TransportInterface() = default;
 
+    //! Copy constructor
+    TransportInterface(
+            const TransportInterface& t) = default;
+
+    //! Copy assignment
+    TransportInterface& operator =(
+            const TransportInterface& t) = default;
+
```

Comment:
I think it makes sense to make transports non-copyable and non-moveable. So declare these as `= delete`, and add move constructor and assignment also as `= delete`.

Reply:
Done in 2bb1c5feceb7be770327e77413667687e841cd7f